### ISSUE TITLE
I have set seed = 0， but the result can not reproduce，why？

### ISSUE THREAD

**Issue by wch243294382** [ultralytics/yolov5#8706]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I have set seed = 0， but the result can not reproduce，why？

**Comment by glenn-jocher** [comment]
@wch243294382 👋 Hello! Thanks for asking about **training reproducibility**. YOLOv5 🚀 uses a single training seed which is set here using the `init_seeds()` function. CPU and Single-GPU trainings should be fully reproducible with `torch>=1.12.0`. Multi-GPU DDP trainings are still not reproducible unfortunately. This is an open issue for us and we could use any help in tracking down this problem.
https://github.com/ultralytics/yolov5/blob/7215a0fb41a90d8a0bf259fa708dff608a1f0262/train.py#L104

This function sets `python`, `numpy` and `torch` seeds and updates `cudnn` settings:
https://github.com/ultralytics/yolov5/blob/7215a0fb41a90d8a0bf259fa708dff608a1f0262/utils/general.py#L198-L214

To set a new training seed for example:
```bash
python train.py --seed 3  # default seed=0
```

Note that even when using the same seed trainings may produce different results, especially when using CUDA backends. See https://pytorch.org/docs/stable/notes/randomness.html for details on factors affecting PyTorch training reproducibility and sources of randomness.

<img width="949" alt="Screen Shot 2022-01-08 at 11 20 59 AM" src="https://user-images.githubusercontent.com/26833433/148660366-9155dfd9-351c-4174-818f-2b1b907b3e4e.png">

Good luck 🍀 and let us know if you have any other questions!
